crates/ruff_python_formatter/src/expression/expr_named_expr.rs

```diff
@@ -46,6 +46,7 @@ impl NeedsParentheses for ExprNamedExpr {
             || parent.is_with_item()
             || parent.is_stmt_delete()
             || parent.is_stmt_for()
+            || parent.is_stmt_function_def()
```

Comment:
For completeness: this change alone was actually not sufficient; we _also_ need https://github.com/astral-sh/ruff/pull/6436, which ensures that we respect "required" parentheses for nodes.